packages/cpp/ArmoniK.Api.Worker/source/Worker/TaskHandler.cpp

```diff
@@ -0,0 +1,400 @@
+using armonik::api::grpc::v1::worker::ProcessRequest;
+using grpc::Channel;
+using grpc::ChannelInterface;
+using grpc::ClientContext;
+using grpc::Status;
+using namespace armonik::api::grpc::v1::agent;
+
+/**
+ * @brief Construct a new Task Handler object
+ *
+ * @param client the agent client
+ * @param request_iterator The request iterator
+ */
+TaskHandler::TaskHandler(std::unique_ptr<Agent::Stub> client,
+                         std::unique_ptr<grpc::ClientReader<ProcessRequest>> request_iterator) {
+  stub_ = std::move(client);
+  request_iterator_ = std::move(request_iterator);
+}
+
+/**
+ * @brief Initialise the task handler
+ *
+ */
+void TaskHandler::init() {
+  ProcessRequest Request;
+  std::unique_ptr<grpc::ClientReader<ProcessRequest>> request_iterator;
+  request_iterator->Read(&Request);
+  if (!request_iterator->Read(&Request)) {
+    throw std::runtime_error("Request stream ended unexpectedly.");
+  }
+
+  if (Request.compute().type_case() != armonik::api::grpc::v1::worker::ProcessRequest_ComputeRequest::kInitRequest) {
+    throw std::runtime_error("Expected a Compute request type with InitRequest to start the stream.");
+  }
+  auto &init_request = Request.compute().init_request();
+  session_id_ = init_request.session_id();
+  task_id_ = init_request.task_id();
+  task_options_ = init_request.task_options();
+  expected_result_ = init_request.expected_output_keys();
+  token_ = Request.communication_token();
+  config_ = init_request.configuration();
+
+  std::vector<std::byte> payload;
+  if (init_request.payload().data_complete()) {
+    std::string temp = init_request.payload().data();
+    payload.resize(temp.length());
+    for (size_t i = 0; i < temp.length(); i++) {
+      payload[i] = static_cast<std::byte>(temp[i]);
+    }
+
+  } else {
+    std::vector<std::string> chuncks;
+    auto datachunck = init_request.payload();
+
+    chuncks.push_back(datachunck.data());
+
+    while (!datachunck.data_complete()) {
+      if (request_iterator->Read(&Request)) {
+        throw std::runtime_error("Request stream ended unexpectedly.");
+      }
+      if (Request.compute().type_case() != armonik::api::grpc::v1::worker::ProcessRequest_ComputeRequest::kPayload) {
+        throw std::runtime_error("Expected a Compute request type with Payload to continue the stream.");
+      }
+
+      datachunck = Request.compute().payload();
+
+      chuncks.push_back(datachunck.data());
+    }
+
+    size_t size = chuncks.size();
+
+    auto payload_data = new std::byte[size];
+    int address = 0;
+    for (auto iter = chuncks.begin(); iter != chuncks.end(); iter++) {
+      std::string temp_str = *iter;
+
+      for (size_t i = 0; i < temp_str.length(); i++) {
+        payload_data[i + address] = static_cast<std::byte>(temp_str[i]);
+      }
+      address += temp_str.length();
+    }
+
+    payload.resize(size);
+    std::copy_n(payload_data, size, std::back_inserter(payload));
+  }

```

Comment:
1 : what's the point of payload_data ? You already have payload
2 : Both payload_data and payload are local, they cannot be accessed !